### PR TITLE
OCP specific ServiceMonitors

### PR DESCRIPTION
**Pull request by bastjan**
Adds an addon `openshift4` to fix the ServiceMonitors from the [`k8s-control-plane.libsonnet` component](https://github.com/prometheus-operator/kube-prometheus/blob/main/jsonnet/kube-prometheus/components/k8s-control-plane.libsonnet) to work with Openshift4.

Precondition for https://github.com/projectsyn/package-monitoring/pull/6.

## Checklist

- [x] PR contains a single logical change (to build a better changelog).
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.

### CODE REVIEW

File: component/addons/openshift4.libsonnet [hunk 1]
```diff
@@ -0,0 +1,102 @@
+// This addon allows this component to be deployed on OpenShift clusters.
+// It:
+// - patches the upstream ServiceMonitors to work with OpenShift.
+// - adds the `remove-securitycontext` addon to remove the security context from deployments.
+
```
Comment (glrf) [inline component/addons/openshift4.libsonnet:5]: I like the idea to introduce a generic `openshift4` addon that will add all necessary other addons. 

Nitpick: I think I would even prefer if we extract the control-plane patching in a separate addon `openshift4-control-plane`. I mean the addon will never be useful on its own but I feel the code would be a bit easier to read.

File: component/addons/openshift4.libsonnet [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      spec+: {
+        endpoints+: [],
+      },
+    },
+
+    serviceMonitorCoreDNS+: {
+      spec+: {
+        endpoints+: [],
+      },
+    },
```
Comment (glrf) [inline component/addons/openshift4.libsonnet:50]: Maybe I'm missing some edge case but why is this needed?

**Review by glrf — APPROVED** [review]
LGTM

File: component/addons/openshift4.libsonnet (hunk 1 above)
Comment (bastjan) [reply component/addons/openshift4.libsonnet:5]: I moved the service monitor into it's own addon 👍

File: component/addons/openshift4.libsonnet (hunk 2 above)
Comment (bastjan) [reply component/addons/openshift4.libsonnet]: It's there so the jsonnet language server shuts up. The language server does not know it extends an upstream component and complains about using `super`. Without having a super object.
